src/app/App.js

```diff
@@ -327,6 +328,11 @@ class App extends EventEmitter {
 			console.log('Link clicked outside app\'s base path');
 			return false;
 		}
+		if (this.isSamePendingNavigationPath_(path)) {
+			console.log('Discarding the navigation because is trying to navigate to the same path');
+			event.preventDefault();
```

Comment:
Given that `event` is a new argument introduced by this PR and that canNavigate is a public API, this call is not safe. [Most certainly](https://github.com/liferay/liferay-portal/search?q=canNavigate&unscoped_q=canNavigate) there is code somewhere calling `canNavigate(path)` without passing that argument.

Also, this function was pure before this change, now it has side effects. I think it would be better to call that `event.preventDefault()` from outside of it if we can.

Reply:
Sorry for delaying, I was testing the move of `isSamePendingNavigationPath_` to `maybeNavigate_` function instead of calling from `canNavigate` and it looks fine. I just noticed when you told that passing `event` as a parameter can be a breaking change since it's a public method. I'll send a update on it :)